contracts/remarkables/src/state.rs

```diff
@@ -0,0 +1,23 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cosmwasm_std::{Addr, Coin};
+use cw_storage_plus::{Item, Map};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct RarityState {
+    pub level: u32,
+    pub mint_fees: Vec<Coin>,
+    pub engagement_threshold: u32,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct ConfigState {
+    pub admin: Addr,
+    pub subspace_id: u64,
+    pub cw721_code_id: u64,
+}
+
+pub const RARITY: Map<u32, RarityState> = Map::new("rarity");
```

Comment:
Can we use a `Vec` instead? If we remove the `level` from `msg::Rarity` and use the index of `InstantiateMsg::rarities` as rarity level we don't need a Map and we can access the rarity levels in O(1)

Reply:
I have same thought to use `Vec` instead of `Map`. Finally, I decided to use `Map` since It forces the rarity level must be sequential and start from 0, and `Map` provides more flexibility if we need to add more rarity level in the future.
But yeah, it makes more sense to use vector because I can't find the use case if the current contract has no needs to add more levels and customize the level number.

Reply:
Cool, changed to `Vec`.